tracer/src/Datadog.Trace/AppSec/ObjectExtractor.cs

```diff
@@ -107,15 +114,15 @@ internal static object Extract(object body)
                     ilGen.Emit(OpCodes.Ret);
                     var func = (Func<object, object>)dynMethod.CreateDelegate(typeof(Func<object, object>));
 
-                    fieldExtractors[i] = new FieldExtractor() { Name = propertyName, Type = field.FieldType, Accessor = func };
+                    fieldExtractors[i] = new FieldExtractor(propertyName!, field.FieldType, func);
```

Comment:
propertyName cant be null as it's tested before
```suggestion
                    fieldExtractors[i] = new FieldExtractor(propertyName, field.FieldType, func);
```

Reply:
Yes, I know, but for some reason, the compiler doesn't take the previous IsNullOrEmpty check into account and I got an error message.

Reply:
Yeah, it's an annoyance of .NET Framework IIRC - the core libraries aren't annotated, so it doesn't recognise `string.IsNullOrEmpty` in the null analysis 🙄